### PR TITLE
243 conference proceeding

### LINKED ISSUE #243: Conference Proceeding

### PR DESCRIPTION
**Pull request by faisalwahyudi10**
- [x] Galley Articles in Submissions
- [x] Add Input Field Custom Filename On Submission Galleys
- [x] Add Input Field Page Article and Cover Image On Submission
- [x] Proceedings Page
- [x] Proceeding Detail Page
- [x] Article Detail Page
- [x] Display Current Published Proceeding and articles in Conference Home
- [x] Preview Proceeding and Article (while at the editing stage)

### CODE REVIEW

**Comment by faisalwahyudi10** [comment]
**- Galley Articles in Submissions**

image

**- Add Input Field Custom Filename On Submission Galleys**

image

**- Add Input Field Page Article and Cover Image On Submission**

image
image

**Comment by faisalwahyudi10** [comment]
**- Proceedings Page**

image

**- Proceeding Detail Page**
image
image

**Comment by faisalwahyudi10** [comment]
**- Article Detail Page**

image
image

**- Display Current Published Proceeding and articles in Conference Home**

[Screencast from 2024-05-18 16-49-35.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/a8111ddb-5d7d-4db7-af6e-fae504bf321f)

**Comment by rahmanramsi** [comment]
Error sewaktu mengakses preview

https://github.com/OpenSynergic/leconfe/assets/48276043/cc24b709-5e02-479a-a61c-22904550a3d6

https://flareapp.io/share/VmeN9GaP

**Comment by rahmanramsi** [comment]
Error sewaktu mengakses preview
CleanShot 2024-05-20 at 11 58 27


https://flareapp.io/share/o7AQRgv7

**Comment by rahmanramsi** [comment]
Halaman article bisa diakses walaupun proceeding belum di publish 
CleanShot 2024-05-20 at 14 13 29

CleanShot 2024-05-20 at 14 14 09

**Comment by rahmanramsi** [comment]
Semisal datanya kosong, seharusnya ada keterangan bahwa belum ada proceeding yg dipublish
CleanShot 2024-05-20 at 14 17 26

**Comment by rahmanramsi** [comment]
Contributor di article tidak responsive
CleanShot 2024-05-20 at 14 18 36

**Comment by rahmanramsi** [comment]
Galley tidak muncul di halaman depan article
CleanShot 2024-05-20 at 14 21 35

CleanShot 2024-05-20 at 14 22 01

**Comment by rahmanramsi** [comment]
Current proceeding seharusnya berada dipaling bawah, bukan Speaker yg berada dipaling bawah 
CleanShot 2024-05-20 at 14 24 01

Gambar cover juga jangan sampai stretch

**Comment by rahmanramsi** [comment]
Proceeding list, Tidak responsive di tampilan mobile

CleanShot 2024-05-20 at 15 18 40

**Comment by rahmanramsi** [comment]
Seharusnya semisal proceeding yg dipublish tidak ada articlenya, tampilkan informasi bahwa tidak ada artikel dipublish diproceeding ini.

CleanShot 2024-05-20 at 15 27 21

**Comment by faisalwahyudi10** [comment]
List Task and Revision
- [x]  Can’t preview if submission is not registered in the proceedings
- [x]  Manage the published date when the data is null in the article preview
- [x]  Rename file name from list-article to article-summary and current-proceeding to proceeding
- [x]  Combine the proceeding component with the article summary
- [x]  Consolidate the article list component into a single data set
- [x]  Don't stretch the cover image
- [x]  Place the proceedings component below the speakers component
- [x]  Responsive mobile
- [x]  A galley appears on the article details page
- [x]  The article page should not be accessible if the proceedings have not been released
- [x]   View Empty state data (proceedings and articles)

**Comment by faisalwahyudi10** [comment]
**- Can’t preview if submission is not registered in the proceedings**

[Screencast from 2024-05-24 08-45-55.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/d0c0d3ff-29c1-4f11-b945-4e71f3f99869)

**- Manage the published date when the data is null in the article preview**

image

**- Rename file name from list-article to article-summary and current-proceeding to proceeding**

https://github.com/OpenSynergic/leconfe/pull/251/commits/c2674f7a65853a88ecd90a95a5fa4462827f2f15
https://github.com/OpenSynergic/leconfe/pull/251/commits/1c3e6d1c3cdfd3cc49ae75cf0c010cc78ae1c455

**- Combine the proceeding component with the article summary**
https://github.com/OpenSynergic/leconfe/pull/251/commits/c2674f7a65853a88ecd90a95a5fa4462827f2f15

**-  Consolidate the article list component into a single data set**
https://github.com/OpenSynergic/leconfe/pull/251/commits/1c3e6d1c3cdfd3cc49ae75cf0c010cc78ae1c455

**- Place the proceedings component below the speakers component**

image

**- View Empty state data (proceedings and articles)**

image
image

**- A galley appears on the article details page**

image

**- The article page should not be accessible if the proceedings have not been released**

[Screencast from 2024-05-24 09-14-23.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/056fe0b5-13ed-48f3-917a-3132bde16500)

**Comment by faisalwahyudi10** [comment]
**- Responsive Mobile**

1. Page Home

[Screencast from 2024-05-24 09-17-40.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/0cfac1ae-82b7-4358-87b2-a79672913d9a)

2. Page List Proceeding

[Screencast from 2024-05-24 09-19-11.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/0dbfd61e-7fbd-4351-8a5a-7ec009b2626a)

3. Proceeding Detail

[Screencast from 2024-05-24 09-22-17.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/5cd7e926-05e6-4274-9d94-3b4e75c4ecbe)

4. Article Detail

[Screencast from 2024-05-24 09-23-43.webm](https://github.com/OpenSynergic/leconfe/assets/121210620/e42b46ac-d8a6-4b51-b2e5-2a4fd2f3482d)

File: app/Frontend/Conference/Pages/SubmissionDetail.php [hunk 1]
```diff
@@ -17,7 +19,37 @@ class SubmissionDetail extends Page
...
+    }
+
+    public function canAccess(): bool
+    {
+        return $this->submission->proceeding && 
+        (
+            (StageManager::editing()->isStageOpen() && auth()->user()?->can('editing', $this->submission)) || 
+            ($this->submission?->isPublished() && $this->submission->proceeding?->isPublished())
+        );
+    }
```
Comment (rahmanramsi) [inline app/Frontend/Conference/Pages/SubmissionDetail.php]: Susah dipahami, breakdown jadi beberapa if agar mudah dipahami

File: resources/views/frontend/conference/pages/home.blade.php
Comment (rahmanramsi) [inline resources/views/frontend/conference/pages/home.blade.php:1]: Bisa dimatii lah plugin yg otomatis mensorting nama class, nama class yg tujuannya gasan dibaca developer gsan memodifikasi tampilan kaya `cf-name`, `cf-topics` jadi paling belakang di class namenya.

**Review by rahmanramsi — CHANGES_REQUESTED** [review]

File: app/Frontend/Conference/Pages/SubmissionDetail.php (hunk 1 above)
Comment (faisalwahyudi10) [reply app/Frontend/Conference/Pages/SubmissionDetail.php]: Kaya gini gimana bang?

    // Cek apakah submission memiliki proceeding
    if (!$this->submission->proceeding) {
        return false;
    }

    // Cek apakah stage editing terbuka dan pengguna memiliki izin untuk mengedit
    if (StageManager::editing()->isStageOpen() && auth()->user()?->can('editing', $this->submission)) {
        return true;
    }

    // Cek apakah submission dan proceeding dipublikasikan
    if ($this->submission->isPublished() && $this->submission->proceeding->isPublished()) {
        return true;
    }

    // Jika tidak memenuhi kondisi apapun di atas, kembalikan false
    return false;

Comment (rahmanramsi) [reply app/Frontend/Conference/Pages/SubmissionDetail.php]: iya, kda ush pakai comment klo coding nya sudah jelas

**Review by rahmanramsi — APPROVED** [review]

**Pull request merged**